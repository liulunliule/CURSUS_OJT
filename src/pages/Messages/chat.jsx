import React from "react";
import "./index.scss";
import { mess1 } from "../../assets";

const Chat = () => {
  const currentChat = [
    {
      text: "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec rutrum congue leo eget malesuada. Vivamus suscipit tortor eget felis porttitor.",
      time: "Sat, April 10, 1:08 PM",
      isOwnMessage: true,
    },
    {
      text: "Cras ultricies ligula.",
      time: "5 minutes ago",
      isOwnMessage: false,
    },
    {
      text: "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec rutrum congue leo eget malesuada. Vivamus suscipit tortor eget felis porttitor.",
      time: "Sat, April 10, 1:08 PM",
      isOwnMessage: true,
    },
    {
      text: "Lorem ipsum dolor sit amet",
      time: "2 minutes ago",
      isOwnMessage: false,
    },
    {
      text: "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec rutrum congue leo eget malesuada. Vivamus suscipit tortor eget felis porttitor.",
      time: "Sat, April 10, 1:08 PM",
      isOwnMessage: true,
    },
    { text: "....", time: "Typing...", isOwnMessage: false },
  ];
  return (
    <>
      <header className="chat-header">
        <img src={mess1} alt="John Doe" />
        <div className="header-content">
          <h3>John Doe</h3>
          <span className="status">ONLINE</span>
        </div>
        <div className="options-menu">
          <div className="ellipsis">...</div>
          <ul className="menu-items">
            <li>Delete</li>
            <li>Block</li>
            <li>Report</li>
            <li>Mute</li>
          </ul>
        </div>
      </header>
      <div className="chat-messages">
        {currentChat.map((msg, index) => (
          <div
            key={index}
            className={`chat-message ${msg.isOwnMessage ? "own" : ""}`}
          >
            <p>{msg.text}</p>
            <span className="time">{msg.time}</span>
          </div>
        ))}
      </div>
      <footer className="chat-input">
        <input type="text" placeholder="Write a message..." />
        <button>&#9658;</button>
      </footer>
    </>
  );
};

export default Chat;
