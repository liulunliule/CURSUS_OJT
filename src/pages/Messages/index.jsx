import React from "react";
import "./index.scss";
import { mess1 } from "../../assets";
import { mess2 } from "../../assets";
import { mess3 } from "../../assets";
import { mess4 } from "../../assets";
import { mess5 } from "../../assets";
import { mess6 } from "../../assets";
import { mess7 } from "../../assets";
import { mess8 } from "../../assets";
import { useSelector } from "react-redux";

const messages = [
  {
    name: "John Doe",
    text: "Hi! Sir, How are you. I ask you one thing pl...",
    time: "7 hours ago",
    avatar: mess1,
    unread: 2,
  },
  {
    name: "Kerstin Cable",
    text: "Hello, I paid you video tutorial but did not...",
    time: "8 hours ago",
    avatar: mess2,
    unread: 3,
  },
  {
    name: "Jose Portilla",
    text: "Thanks Sir, Such a nice video.",
    time: "15 hours ago",
    avatar: mess3,
    unread: 1,
  },
  {
    name: "Farhat Amin",
    text: "Hi! Sir, this is a purchase key CFKX12536...",
    time: "22 hours ago",
    avatar: mess4,
    unread: 7,
  },
  {
    name: "Kyle Pew",
    text: "Pls! Upload .NET Course",
    time: "2 days ago",
    avatar: mess5,
    unread: 12,
  },
  {
    name: "Eli Natoli",
    text: "Hi Sir thank you very very much for bootst...",
    time: "5 days ago",
    avatar: mess6,
  },
  {
    name: "Jaysen Batchelor",
    text: "Thank you! Sir",
    time: "7 days ago",
    avatar: mess7,
    unread: 6,
  },
  {
    name: "Quinton Batchelor",
    text: "Hey!",
    time: "13 days ago",
    avatar: mess8,
    unread: 1,
  },
];

const currentChat = [
  { text: "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec rutrum congue leo eget malesuada. Vivamus suscipit tortor eget felis porttitor.", time: "Sat, April 10, 1:08 PM", isOwnMessage: true },
  { text: "Cras ultricies ligula.", time: "5 minutes ago", isOwnMessage: false },
  { text: "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec rutrum congue leo eget malesuada. Vivamus suscipit tortor eget felis porttitor.", time: "Sat, April 10, 1:08 PM", isOwnMessage: true },
  { text: "Lorem ipsum dolor sit amet", time: "2 minutes ago", isOwnMessage: false },
  { text: "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec rutrum congue leo eget malesuada. Vivamus suscipit tortor eget felis porttitor.", time: "Sat, April 10, 1:08 PM", isOwnMessage: true },
  { text: "....", time: "Typing...", isOwnMessage: false },
];

const Messages = () => {
  const isShowAll = useSelector((state) => state.savedCourse.isShowAll);
  return (
    <div className="chat-tab">
      <header className="chatting-header">
        <h2>Messages</h2>
      </header>
      <div className={`chat-app ${isShowAll ? "active" : ""}`}>
        <aside className="message-list">
          <input type="text" placeholder="Search Messages..." />
          <ul>
            {messages.map((msg, index) => (
              <li key={index} className={`message-item ${index === 0 ? "first" : ""}`}>
                <img src={msg.avatar} alt={msg.name} />
                <div className="message-info">
                  <h3>{msg.name}</h3>
                  <p>{msg.text}</p>
                </div>
                <div className="time-unread-wrapper">
                  <span className="time">{msg.time}</span>
                  {msg.unread > 0 && <span className="unread-count">{msg.unread}</span>}
                </div>
              </li>
            ))}
          </ul>
        </aside>
        <main className="chat-window">
          <header className="chat-header">
            <img src={mess1} alt="John Doe" />
            <div className="header-content">
              <h3>John Doe</h3>
              <span className="status">ONLINE</span>
            </div>
            <div className="options-menu">
              <div className="ellipsis">...</div>
              <ul className="menu-items">
                <li>Delete</li>
                <li>Block</li>
                <li>Report</li>
                <li>Mute</li>
              </ul>
            </div>
          </header>
          <div className="chat-messages">
            {currentChat.map((msg, index) => (
              <div key={index} className={`chat-message ${msg.isOwnMessage ? "own" : ""}`}>
                <p>{msg.text}</p>
                <span className="time">{msg.time}</span>
              </div>
            ))}
          </div>
          <footer className="chat-input">
            <input type="text" placeholder="Write a message..." />
            <button>&#9658;</button>
          </footer>
        </main>
      </div>
    </div>
  );
};

export default Messages;