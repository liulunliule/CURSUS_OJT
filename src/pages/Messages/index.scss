.chat-tab {
  margin-top: 50px;
  background-color: #f8f8f8;
  padding: 20px;
  height: 100%;
  display: flex;
  flex-direction: column;

  .chatting-header {
    padding: 20px;

    h2 {
      margin-left: 270px;
      font-size: 20px;
    }
  }
}

.chat-app {
  display: flex;
  height: 100%;
  background-color: #f0f0f0;
  margin: 0;
  font-family: Arial, sans-serif;
  margin-top: 3px;
  margin-left: 280px;
  margin-right: 45px;
  margin-bottom: 40px;
  transition: margin-left 0.5s ease;

  &.active {
    transition: margin-left 0.5s ease;
    margin-left: 0;
  }

  .message-list {
    width: 500px;
    background-color: #fff;
    border-right: 1px solid #e0e0e0;

    input {
      width: 95%;  
      padding: 9px;
      margin-left: 10px;
      margin-top: 10px;
      margin-bottom: 10px;
      border: 1px solid #e0e0e0;
      border-radius: 5px;
    }

    ul {
      list-style: none;
      padding: 0;
      margin: 0;

      li.message-item {
        display: flex;
        align-items: center;
        padding: 10px 20px;
        border-bottom: 1px solid #e0e0e0;
        cursor: pointer;
        background-color: #fff;

        &.first {
          background-color: #ffe3e3;
        }

        &:not(.first):hover {
          background-color: #f9f9f9;
        }

        img {
          width: 40px;
          height: 40px;
          border-radius: 50%;
          margin-right: 10px;
        }

        .message-info {
          flex-grow: 1;
          display: flex;
          flex-direction: column;

          h3 {
            margin: 0;
            font-size: 14px;
            color: #333;
          }

          p {
            margin: 0;
            font-size: 12px;
            color: #888;
          }
        }

        .time-unread-wrapper {
          display: flex;
          align-items: center;
          margin-left: 20px;

          .time {
            font-size: 12px;
            color: #888;
            margin-right: 10px;
          }

          .unread-count {
            background-color: red;
            color: #fff;
            border-radius: 50%;
            width: 20px;
            height: 20px;
            display: flex;
            align-items: center;
            justify-content: center;
            font-size: 12px;
          }
        }
      }
    }
  }

  .chat-window {
    flex-grow: 1;
    display: flex;
    flex-direction: column;
    overflow: hidden;

    .chat-header {
      display: flex;
      align-items: center;
      padding: 10px 20px;
      background-color: #fff;
      border-bottom: 1px solid #e0e0e0;

      img {
        width: 50px;
        height: 50px;
        border-radius: 50%;
        margin-right: 10px;
      }

      .header-content {
        display: flex;
        flex-direction: column;
      }

      h3 {
        margin: 0;
        font-size: 18px;
      }

      .status {
        background-color: rgb(1, 215, 1);
        text-align: center;
        color: #fff;
        padding: 1px 1px;
        border-radius: 5px;
        font-size: 12px;
        font-weight: bold;
      }

      .options-menu {
        position: relative;
        cursor: pointer;
        margin-left: auto;

        .ellipsis {
          font-size: 20px;
          color: #888;
        }

        .menu-items {
          position: absolute;
          right: 0;
          background-color: #fff;
          box-shadow: 0px 0px 10px rgba(0, 0, 0, 0.1);
          border-radius: 5px;
          padding: 20px;
          display: none;
          z-index: 1;

          li {
            list-style: none;
            padding: 8px 0;
            cursor: pointer;

            &:hover {
              background-color: #ffe3e3;
            }
          }
        }

        &:hover {
          .menu-items {
            display: block;
          }
        }
      }
    }

    .chat-messages {
      flex-grow: 1;
      padding: 20px;
      background-color: #fff;
      overflow-y: auto;
      scrollbar-width: none;
      -ms-overflow-style: none;

      &::-webkit-scrollbar {
        width: 0px;
        background: transparent;
      }

      .chat-message {
        margin-bottom: 20px;
        display: flex;
        flex-direction: column;

        &.own {
          align-items: flex-end;

          p {
            background-color: red;
            color: #fff;
            border-radius: 10px 10px 0 10px;
            padding: 10px;
          }
        }

        &:not(.own) {
          align-items: flex-start;

          p {
            background-color: #ffe3e3;
            color: #333;
            border-radius: 10px 10px 10px 0;
            padding: 10px;
          }
        }

        p {
          margin: 0;
          max-width: 70%;
        }

        .time {
          font-size: 12px;
          color: #888;
          margin-top: 5px;
        }
      }
    }

    .chat-input {
      display: flex;
      padding: 20px;
      background-color: #fff;
      border-top: 1px solid #e0e0e0;

      input {
        flex-grow: 1;
        padding: 10px;
        border: 1px solid #e0e0e0;
        border-radius: 5px;
      }

      button {
        background-color: red;
        border: none;
        color: #fff;
        padding: 10px 20px;
        margin-left: 10px;
        border-radius: 5px;
        cursor: pointer;
        font-size: 18px;

        &:hover {
          background-color: black;
        }
      }
    }
  }
}