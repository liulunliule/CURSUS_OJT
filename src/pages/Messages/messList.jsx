import React from "react";
import {
  mess1,
  mess2,
  mess3,
  mess4,
  mess5,
  mess6,
  mess7,
  mess8,
} from "../../assets";
import "./index.scss";

function messList() {
  const messages = [
    {
      name: "John Doe",
      text: "Hi! Sir, How are you. I ask you one thing pl...",
      time: "7 hours ago",
      avatar: mess1,
      unread: 2,
    },
    {
      name: "Kerstin Cable",
      text: "Hello, I paid you video tutorial but did not...",
      time: "8 hours ago",
      avatar: mess2,
      unread: 3,
    },
    {
      name: "Jose Portilla",
      text: "Thanks Sir, Such a nice video.",
      time: "15 hours ago",
      avatar: mess3,
      unread: 1,
    },
    {
      name: "Farhat Amin",
      text: "Hi! Sir, this is a purchase key CFKX12536...",
      time: "22 hours ago",
      avatar: mess4,
      unread: 7,
    },
    {
      name: "Kyle Pew",
      text: "Pls! Upload .NET Course",
      time: "2 days ago",
      avatar: mess5,
      unread: 12,
    },
    {
      name: "Eli Natoli",
      text: "Hi Sir thank you very very much for bootst...",
      time: "5 days ago",
      avatar: mess6,
    },
    {
      name: "Jaysen Batchelor",
      text: "Thank you! Sir",
      time: "7 days ago",
      avatar: mess7,
      unread: 6,
    },
    {
      name: "Quinton Batchelor",
      text: "Hey!",
      time: "13 days ago",
      avatar: mess8,
      unread: 1,
    },
  ];

  return (
    <>
      <aside className="message-list">
        <input
          type="text"
          className="add mess"
          placeholder="Search Messages..."
        />
        <ul>
          {messages.map((msg, index) => (
            <li
              key={index}
              className={`message-item ${index === 0 ? "first" : ""}`}
            >
              <img src={msg.avatar} alt={msg.name} />
              <div className="message-info">
                <h3>{msg.name}</h3>
                <p>{msg.text}</p>
              </div>
              <div className="time-unread-wrapper">
                <span className="time">{msg.time}</span>
                {msg.unread && (
                  <span className="unread-count">{msg.unread}</span>
                )}
              </div>
            </li>
          ))}
        </ul>
      </aside>
    </>
  );
}

export default messList;
