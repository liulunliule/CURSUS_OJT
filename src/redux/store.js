import { combineReducers, configureStore } from "@reduxjs/toolkit";
import savedCourseReducer from "./features/savedCourseSlice";
import blogReducer from './features/ourBlogSlice';
import officesReducer from './features/officesSlice';
import pressReducer from './features/pressSlice';
import payoutReducer from './features/payoutSlice';
import creditsReducer from "./features/creditsSlice";
import certificateReducer from "./features/myCertificateSlice";
import myProfileReducer from "./features/myProfileSlice";
import myHeaderReducer from "./features/myHeaderSlice";
import mySearchReducer from "./features/mySearchSlice";
import myStudioReducer from "./features/myStudioSlice";
import allInstructorReducer from "./features/allInstructorSlice";
import reviewFeedbackReducer from "./features/reviewfeedbackSlice";
import feedbackReportReducer from "./features/feedbackvsReportSlice";
import notificationsSettingReducer from "./features/NotificationsSettingSlice";
import privacySettingReducer from "./features/PrivacySettingSlice";
import userSlice from "./features/userSlice";
import certificationTestPageReducer from "./features/certificationTestPageSlice";
import certificationTestResultReducer from "./features/certificationTestResultSlice";
import analyticsReducer from "./features/analyticsSlice";
import storage from "redux-persist/lib/storage";
import { persistReducer, persistStore } from "redux-persist";
import courseSlice from "./features/courseSlice";
import checkoutReducer from './features/checkoutSlice';
import earningReducer from './features/earningSlice';
import statementReducer from './features/statementSlice';
import shoppingCartReducer from './features/shoppingCartSlice';

const persistConfig = {
    key: "root",
    storage,
    whitelist: [
        "user",
        "savedCourse",
        "certificate",
        "course",
        "myProfile",
        "myHeader",
        "mySearch",
        "myStudio",
        "allInstructor",
        "reviewFeedback",
        "feedbackReport",
        "notificationsSetting",
        "privacySetting",
    ],
};

const rootReducer = combineReducers({
    user: userSlice,
    savedCourse: savedCourseReducer,
    certificate: certificateReducer,
    certificationTestPage: certificationTestPageReducer,
    certificationTestResult: certificationTestResultReducer,
    analytics: analyticsReducer,
    course: courseSlice,
    myProfile: myProfileReducer,
    myHeader: myHeaderReducer,
    mySearch: mySearchReducer,
    myStudio: myStudioReducer,
    allInstructor: allInstructorReducer,
    reviewFeedback: reviewFeedbackReducer,
    feedbackReport: feedbackReportReducer,
    checkout: checkoutReducer,
    earnings: earningReducer,
    statement: statementReducer,
    credits: creditsReducer,
    payout: payoutReducer,
    blogs: blogReducer,
    offices: officesReducer,
    press: pressReducer,
    notificationsSetting: notificationsSettingReducer,
    privacySetting: privacySettingReducer,
    shoppingCart: shoppingCartReducer,
});

const persistedReducer = persistReducer(persistConfig, rootReducer);

export const store = configureStore({
    reducer: persistedReducer,
});

export const persistor = persistStore(store);
