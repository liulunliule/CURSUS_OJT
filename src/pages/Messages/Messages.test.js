import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import Messages from './index';

const initialState = {
  savedCourse: {
    isShowAll: false,
  },
};

const reducer = (state = initialState, action) => state;
const store = createStore(reducer);

test('renders Messages component correctly', () => {
  const { asFragment } = render(
    <Provider store={store}>
      <Messages />
    </Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});