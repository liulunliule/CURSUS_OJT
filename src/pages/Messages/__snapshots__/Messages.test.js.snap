// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Messages component correctly 1`] = `
<DocumentFragment>
  <div
    class="chat-tab"
  >
    <header
      class="chatting-header"
    >
      <h2>
        Messages
      </h2>
    </header>
    <div
      class="chat-app "
    >
      <aside
        class="message-list"
      >
        <input
          class="add mess"
          placeholder="Search Messages..."
          type="text"
        />
        <ul>
          <li
            class="message-item first"
          >
            <img
              alt="John Doe"
              src="mess1.jpg"
            />
            <div
              class="message-info"
            >
              <h3>
                John Doe
              </h3>
              <p>
                Hi! Sir, How are you. I ask you one thing pl...
              </p>
            </div>
            <div
              class="time-unread-wrapper"
            >
              <span
                class="time"
              >
                7 hours ago
              </span>
              <span
                class="unread-count"
              >
                2
              </span>
            </div>
          </li>
          <li
            class="message-item "
          >
            <img
              alt="Kerstin Cable"
              src="mess2.jpg"
            />
            <div
              class="message-info"
            >
              <h3>
                Kerstin Cable
              </h3>
              <p>
                Hello, I paid you video tutorial but did not...
              </p>
            </div>
            <div
              class="time-unread-wrapper"
            >
              <span
                class="time"
              >
                8 hours ago
              </span>
              <span
                class="unread-count"
              >
                3
              </span>
            </div>
          </li>
          <li
            class="message-item "
          >
            <img
              alt="Jose Portilla"
              src="mess3.jpg"
            />
            <div
              class="message-info"
            >
              <h3>
                Jose Portilla
              </h3>
              <p>
                Thanks Sir, Such a nice video.
              </p>
            </div>
            <div
              class="time-unread-wrapper"
            >
              <span
                class="time"
              >
                15 hours ago
              </span>
              <span
                class="unread-count"
              >
                1
              </span>
            </div>
          </li>
          <li
            class="message-item "
          >
            <img
              alt="Farhat Amin"
              src="mess4.jpg"
            />
            <div
              class="message-info"
            >
              <h3>
                Farhat Amin
              </h3>
              <p>
                Hi! Sir, this is a purchase key CFKX12536...
              </p>
            </div>
            <div
              class="time-unread-wrapper"
            >
              <span
                class="time"
              >
                22 hours ago
              </span>
              <span
                class="unread-count"
              >
                7
              </span>
            </div>
          </li>
          <li
            class="message-item "
          >
            <img
              alt="Kyle Pew"
              src="mess5.jpg"
            />
            <div
              class="message-info"
            >
              <h3>
                Kyle Pew
              </h3>
              <p>
                Pls! Upload .NET Course
              </p>
            </div>
            <div
              class="time-unread-wrapper"
            >
              <span
                class="time"
              >
                2 days ago
              </span>
              <span
                class="unread-count"
              >
                12
              </span>
            </div>
          </li>
          <li
            class="message-item "
          >
            <img
              alt="Eli Natoli"
              src="mess6.jpg"
            />
            <div
              class="message-info"
            >
              <h3>
                Eli Natoli
              </h3>
              <p>
                Hi Sir thank you very very much for bootst...
              </p>
            </div>
            <div
              class="time-unread-wrapper"
            >
              <span
                class="time"
              >
                5 days ago
              </span>
            </div>
          </li>
          <li
            class="message-item "
          >
            <img
              alt="Jaysen Batchelor"
              src="mess7.jpg"
            />
            <div
              class="message-info"
            >
              <h3>
                Jaysen Batchelor
              </h3>
              <p>
                Thank you! Sir
              </p>
            </div>
            <div
              class="time-unread-wrapper"
            >
              <span
                class="time"
              >
                7 days ago
              </span>
              <span
                class="unread-count"
              >
                6
              </span>
            </div>
          </li>
          <li
            class="message-item "
          >
            <img
              alt="Quinton Batchelor"
              src="mess8.jpg"
            />
            <div
              class="message-info"
            >
              <h3>
                Quinton Batchelor
              </h3>
              <p>
                Hey!
              </p>
            </div>
            <div
              class="time-unread-wrapper"
            >
              <span
                class="time"
              >
                13 days ago
              </span>
              <span
                class="unread-count"
              >
                1
              </span>
            </div>
          </li>
        </ul>
      </aside>
      <main
        class="chat-window"
      >
        <header
          class="chat-header"
        >
          <img
            alt="John Doe"
            src="mess1.jpg"
          />
          <div
            class="header-content"
          >
            <h3>
              John Doe
            </h3>
            <span
              class="status"
            >
              ONLINE
            </span>
          </div>
          <div
            class="options-menu"
          >
            <div
              class="ellipsis"
            >
              ...
            </div>
            <ul
              class="menu-items"
            >
              <li>
                Delete
              </li>
              <li>
                Block
              </li>
              <li>
                Report
              </li>
              <li>
                Mute
              </li>
            </ul>
          </div>
        </header>
        <div
          class="chat-messages"
        >
          <div
            class="chat-message own"
          >
            <p>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec rutrum congue leo eget malesuada. Vivamus suscipit tortor eget felis porttitor.
            </p>
            <span
              class="time"
            >
              Sat, April 10, 1:08 PM
            </span>
          </div>
          <div
            class="chat-message "
          >
            <p>
              Cras ultricies ligula.
            </p>
            <span
              class="time"
            >
              5 minutes ago
            </span>
          </div>
          <div
            class="chat-message own"
          >
            <p>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec rutrum congue leo eget malesuada. Vivamus suscipit tortor eget felis porttitor.
            </p>
            <span
              class="time"
            >
              Sat, April 10, 1:08 PM
            </span>
          </div>
          <div
            class="chat-message "
          >
            <p>
              Lorem ipsum dolor sit amet
            </p>
            <span
              class="time"
            >
              2 minutes ago
            </span>
          </div>
          <div
            class="chat-message own"
          >
            <p>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec rutrum congue leo eget malesuada. Vivamus suscipit tortor eget felis porttitor.
            </p>
            <span
              class="time"
            >
              Sat, April 10, 1:08 PM
            </span>
          </div>
          <div
            class="chat-message "
          >
            <p>
              ....
            </p>
            <span
              class="time"
            >
              Typing...
            </span>
          </div>
        </div>
        <footer
          class="chat-input"
        >
          <input
            placeholder="Write a message..."
            type="text"
          />
          <button>
            â–º
          </button>
        </footer>
      </main>
    </div>
  </div>
</DocumentFragment>
`;
